taf/utils.py

```diff
@@ -388,6 +367,42 @@ def get_file_details(
     return file_length, file_hashes
 
 
+def ensure_pre_push_hook(auth_repo_path: Path) -> bool:
+    hooks_dir = auth_repo_path / ".git" / "hooks"
+    hooks_dir.mkdir(parents=True, exist_ok=True)
+    pre_push_script = hooks_dir / "pre-push"
+    resources_pre_push_script = (
+        Path(__file__).parent / "resources" / "pre-push"
+    ).resolve()
+
+    if not pre_push_script.exists():
+        if not resources_pre_push_script.exists():
+            taf_logger.error(
+                f"Resources pre-push script not found at {resources_pre_push_script}"
+            )
+            return False
+
+        shutil.copy(resources_pre_push_script, pre_push_script)
+        try:
+            if platform.system() != "Windows":
+                # Unix-like systems
+                pre_push_script.chmod(0o755)
+        except Exception as e:
+            taf_logger.error(f"Error setting executable permission: {e}")
+            return False
+
+        # Check if permissions were set correctly on Unix-like systems
+        if platform.system() != "Windows" and not os.access(pre_push_script, os.X_OK):
+            taf_logger.error(
+                f"Failed to set pre-push git hook executable permission. Please set it manually for {pre_push_script}."
+            )
+            return False
+        taf_logger.info("Pre-push hook not present. Pre-push hook added successfully.")
+        return True
+
+    return False
```

Comment:
I think that we should return `True` if the hook already exists, since `False` implies that an error occurred